core/services/pipeline/runner.go

```diff
@@ -287,16 +362,26 @@ func (r *runner) executeTaskRun(ctx context.Context, spec Spec, taskRun *memoryT
 	}
 	l.Debugw("Pipeline task completed", loggerFields...)
 
+	now := time.Now()
+
+	var id uuid.UUID
+	if taskRun.task.Type() == TaskTypeBridge {
+		id = taskRun.task.(*BridgeTask).id
+	} else {
+		id = uuid.NewV4()
```

Comment:
Ah, I see, it's generated here. Neat.

Reply:
Yeah it's set early for the bridge task so we can send it through to the external adapter. Could maybe make it a field on BaseTask and pre-generate for all the task types.